relay-metrics/src/aggregator.rs

```diff
@@ -565,13 +565,14 @@ impl fmt::Debug for CostTracker {
 /// Since this uses an approximate function to estimate the size of buckets, the actual serialized
 /// payload may exceed the size. The estimation function is built in a way to guarantee the same
 /// order of magnitude.
-struct CappedBucketIter<T: Iterator<Item = Bucket>> {
+pub struct CappedBucketIter<T: Iterator<Item = Bucket>> {
```

Comment:
Please apply recursively: this is an exclusive helper of process_batches AFAICT, so it should also move to the service module.